cpp/src/io/orc/stripe_enc.cu

```diff
@@ -1368,6 +1402,27 @@ std::optional<writer_compression_statistics> CompressOrcDataStreams(
   }
 }
 
+void decimal_sizes_to_offsets(device_2dspan<rowgroup_rows const> rg_bounds,
```

Comment:
I can't figure out where is the return object? Or the output is overwritten into this `rg_bounds`? Is there any memory allocation that needs `mr`?

Reply:
`elem_sizes` is the output here - we're doing a scan of the element sizes (within each rowgroup) to get the offset in the output stream.
Maybe I should rename to something like `scan_decimal_sizes`?

Reply:
Is it though? I don't see a sync back from the gpu data in `d_sizes`. Am I missing the trick?

Reply:
`h_sizes`/`d_sizes` hold `device_span`s of the `device_uvector`s in `elem_sizes`. It's just a mutable view that the kernel uses to convert the sizes in `elem_sizes` into offsets within row groups, using the scan.
`rg_bounds` contain the row bound of each row group (for each column). We don't change this here, just use it correctly scan within row groups.

I'll take these questions as a sign that the code is very unclear :(
Some comments are in order, it seems.

Reply:
added comments in a few places